osf/models/node.py

```diff
@@ -441,12 +441,25 @@ def is_preprint(self):
         if not self.preprint_file_id or not self.is_public:
             return False
         if self.preprint_file.node_id == self.id:
-            return self.has_published_preprint
+            return self.has_published_preprint or self.has_pending_rejected_preprint
         else:
             self._is_preprint_orphan = True
             return False
 
     @property
+    def has_pending_rejected_preprint(self):
+        return self.preprints.filter(Q(reviews_state='pending') | Q(reviews_state='rejected')).exists()
+
+    @property
+    def has_preprint_moderated(self):
+        return self.preprints.filter(provider__reviews_workflow__isnull=False).exists()
```

Comment:
Do we want to show the banner for unsubmitted (abandoned) preprints, or should this filter out `initial` state?